bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleRegistryImpl.java

```diff
@@ -154,7 +154,7 @@ protected void unsetModuleTypeRegistry(ModuleTypeRegistry moduleTypeRegistry) {
      *
      * @param templateRegistry templateRegistry service.
      */
-    protected void setTemplateRegistry(RuleTemplateRegistry templateRegistry) {
+    protected void setTemplateRegistry(TemplateRegistry<RuleTemplate> templateRegistry) {
```

Comment:
We don't have other type template registries, but anyway I'll add check for this.